lux/lib/lux/prisms/discord/messages/bulk_delete_messages_prism.ex

```diff
@@ -0,0 +1,155 @@
+defmodule Lux.Prisms.Discord.Messages.BulkDeleteMessagesPrism do
+  @moduledoc """
+  A prism for bulk deleting messages in a Discord channel.
+  Messages must be between 2 weeks old and 2 seconds old to be deleted.
+  Can delete between 2 and 100 messages at once.
+
+  ## Examples
+      iex> BulkDeleteMessagesPrism.handler(%{
+      ...>   channel_id: "123456789",
+      ...>   message_ids: ["987654321", "987654322", "987654323"]
+      ...> }, %{agent: %{name: "Agent"}})
+      {:ok, %{deleted: true, count: 3}}
+  """
+
+  use Lux.Prism,
+    name: "Bulk Delete Discord Messages",
+    description: "Deletes multiple messages in a Discord channel at once",
+    input_schema: %{
+      type: :object,
+      properties: %{
+        channel_id: %{
+          type: :string,
+          description: "The ID of the channel containing the messages",
+          pattern: "^[0-9]{17,20}$"
+        },
+        message_ids: %{
+          type: :array,
+          description: "List of message IDs to delete (2-100 messages, not older than 2 weeks)",
+          items: %{
+            type: :string,
+            pattern: "^[0-9]{17,20}$"
+          },
+          minItems: 2,
+          maxItems: 100
+        },
+        reason: %{
+          type: :string,
+          description: "The reason for deleting the messages (appears in audit log)",
+          maxLength: 512
+        }
+      },
+      required: ["channel_id", "message_ids"]
+    },
+    output_schema: %{
+      type: :object,
+      properties: %{
+        deleted: %{
+          type: :boolean,
+          description: "Whether the messages were successfully deleted"
+        },
+        count: %{
+          type: :integer,
+          description: "Number of messages deleted"
+        }
+      },
+      required: ["deleted", "count"]
+    }
+
+  alias Lux.Lenses.DiscordLens
+  require Logger
+
+  # Discord API error codes
+  @discord_errors %{
+    10003 => "Unknown channel",
+    10008 => "Unknown message",
+    50001 => "Missing access",
+    50013 => "Missing permissions",
+    50034 => "Messages provided are too old for bulk deletion",
+    50035 => "Invalid form body",
+    50021 => "Cannot execute action on this channel type"
+  }
+
+  @doc """
+  Handles the request to bulk delete messages in a Discord channel.
+  """
+  def handler(%{channel_id: channel_id, message_ids: message_ids} = input, %{agent: agent} = _ctx) do
+    Logger.info("Agent #{agent.name} bulk deleting #{length(message_ids)} messages in channel #{channel_id}")
+
+    headers = if input[:reason], do: [{"X-Audit-Log-Reason", input.reason}], else: []
+
+    case DiscordLens.focus(%{
```

Comment:
No please let's not do this. Lenses are intended to be used by Agents, or as building blocs inside Beams. They represent a way to load "resources" (something that other framework such as MCP also try to model separately from "Actions"). If we need a discord client to use inside Prisms, let's just write one (or use a library if it is much better, but I think that for our use case we can easily build a basic discord HTTP client with Req.